BLangNodeTransformer crashes when there's invalid unicode escape sequence

**Description:**
Compilation crashes when there's invalid unicode escape sequence.

**Source:**
```ballerina
const string ESCAPE_PREFIX = "\u{001B[";
```

**Error:**
```
[Error - 11:20:53 AM] Operation 'text/codeAction' failed! [13:40]- [13:40], error: 'Unable to parse unicode value: {001'}
java.lang.IllegalArgumentException: Unable to parse unicode value: {001
	at org.apache.commons.lang3.text.translate.UnicodeUnescaper.translate(UnicodeUnescaper.java:59)
	at org.apache.commons.lang3.text.translate.AggregateTranslator.translate(AggregateTranslator.java:55)
	at org.apache.commons.lang3.text.translate.CharSequenceTranslator.translate(CharSequenceTranslator.java:90)
	at org.apache.commons.lang3.text.translate.CharSequenceTranslator.translate(CharSequenceTranslator.java:64)
	at org.apache.commons.lang3.StringEscapeUtils.unescapeJava(StringEscapeUtils.java:526)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createSimpleLiteral(BLangNodeTransformer.java:5207)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createSimpleLiteral(BLangNodeTransformer.java:5113)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createSimpleLiteral(BLangNodeTransformer.java:5101)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createActionOrExpression(BLangNodeTransformer.java:4917)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createExpression(BLangNodeTransformer.java:4912)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:760)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:472)
	at io.ballerina.compiler.syntax.tree.ConstantDeclarationNode.apply(ConstantDeclarationNode.java:75)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:589)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:472)
	at io.ballerina.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.accept(BLangNodeTransformer.java:502)
	at io.ballerina.projects.DocumentContext.compilationUnit(DocumentContext.java:111)
	at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:356)
	at io.ballerina.projects.ModuleCompilationState$3.compile(ModuleCompilationState.java:100)
	at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:323)
	at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:168)
	at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:160)
	at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:96)
	at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:206)
	at io.ballerina.projects.Package.getCompilation(Package.java:130)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.waitAndGetPackageCompilation(BallerinaWorkspaceManager.java:217)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.semanticModel(BallerinaWorkspaceManager.java:196)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.getAvailableCodeActions(CodeActionRouter.java:65)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:46)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:34)
	at org.ballerinalang.langserver.LangExtensionDelegator.codeActions(LangExtensionDelegator.java:133)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeAction$8(BallerinaTextDocumentService.java:335)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NumberFormatException: For input string: "{001"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:652)
	at org.apache.commons.lang3.text.translate.UnicodeUnescaper.translate(UnicodeUnescaper.java:56)
	... 39 more
```
**Affected Versions:**

--- a/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java
+++ b/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java
@@ -5183,6 +5183,7 @@ public class BLangNodeTransformer extends NodeTransformer<BLangNode> {
                 }
             }
             String originalText = text; // to log the errors
+            Location pos = getPosition(literal);
             Matcher matcher = IdentifierUtils.UNICODE_PATTERN.matcher(text);
             int position = 0;
             while (matcher.find(position)) {
@@ -5192,7 +5193,6 @@ public class BLangNodeTransformer extends NodeTransformer<BLangNode> {
                         || hexDecimalVal > Constants.MAX_UNICODE) {
                     String hexStringWithBraces = matcher.group(0);
                     int offset = originalText.indexOf(hexStringWithBraces) + 1;
-                    Location pos = getPosition(literal);
                     dlog.error(new BLangDiagnosticLocation(currentCompUnitName,
                                     pos.lineRange().startLine().line(),
                                     pos.lineRange().endLine().line(),
@@ -5205,7 +5205,11 @@ public class BLangNodeTransformer extends NodeTransformer<BLangNode> {
                 matcher = IdentifierUtils.UNICODE_PATTERN.matcher(text);
             }
             if (type != SyntaxKind.TEMPLATE_STRING && type != SyntaxKind.XML_TEXT_CONTENT) {
-                text = StringEscapeUtils.unescapeJava(text);
+                try {
+                    text = StringEscapeUtils.unescapeJava(text);
+                } catch (Exception e) {
+                    dlog.error(pos, DiagnosticErrorCode.INVALID_UNICODE, originalText);
+                }
             }
 
             typeTag = TypeTags.STRING;

--- a/tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/string/UnicodeNegativeTest.java
+++ b/tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/string/UnicodeNegativeTest.java
@@ -45,6 +45,8 @@ public class UnicodeNegativeTest {
         BAssertUtil.validateError(compileResult, index++, "invalid unicode '\\u{DAFF}'", 24, 46);
         BAssertUtil.validateError(compileResult, index++, "invalid unicode '\\u{12FFFF}'", 25, 18);
         BAssertUtil.validateError(compileResult, index++, "invalid unicode '\\u{DFFFAAA}'", 25, 33);
-        BAssertUtil.validateError(compileResult, index, "invalid unicode '\\u{FFFFFFF}'", 25, 49);
+        BAssertUtil.validateError(compileResult, index++, "invalid unicode '\\u{FFFFFFF}'", 25, 49);
+        BAssertUtil.validateError(compileResult, index++, "invalid string numeric escape sequence", 26, 17);
+        BAssertUtil.validateError(compileResult, index, "invalid unicode '\\u{001B['", 26, 17);
     }
 }

--- a/tests/jballerina-unit-test/src/test/resources/test-src/types/string/unicode-negative.bal
+++ b/tests/jballerina-unit-test/src/test/resources/test-src/types/string/unicode-negative.bal
@@ -23,4 +23,5 @@ function testUnicodeNegative() {
     string s6 = "\u{DFFF}\u{DAFF}";
     string s7 = "\u{12FFFF} ABC \u{DFFF} DEF \u{DAFF}";
     string s8 = "\u{12FFFF} ABC \u{DFFFAAA} DEF \u{FFFFFFF}";
+    string s9 = "\u{001B[";
 }

